lib/streamlit/runtime/pages_manager.py

```diff
@@ -291,7 +292,7 @@ def set_pages(self, pages: dict[PageHash, PageInfo]) -> None:
 
         self.pages_strategy.set_pages(pages)
         self._cached_pages = pages
-        self._on_pages_changed.send()
+        # We deliberately don't notify on page change as V2 does not require of this.
```

Comment:
I would just get rid of this comment if we're getting rid of the call to `self._on_pages_changed.send()` in `get_pages` as well